runbooks/workflow/creating-maintaining-littlepay-data-syncs.md

```diff
@@ -0,0 +1,51 @@
+
+When you receive an access key from Littlepay for an agency that is being newly synced, you need to add that access key to Google Cloud Secret Manager, named with the format `LITTLEPAY_AWS_IAM_<MERCHANT_ID>_ACCESS_KEY` like the existing keys. Secret names are uppercase, and we have removed hyphens (or convert to underscore) in the `merchant_id` historically.
+
+Next, [create a service account in the cal-itp-data-infra project](https://console.cloud.google.com/iam-admin/serviceaccounts/create?walkthrough_id=iam--create-service-account&project=cal-itp-data-infra#step_index=1) specifically for this agency's Littlepay sync, using the naming convention `<MERCHANT_ID>-payments-user`. There is a pre-configured role titled Agency Payments Service User, which will give the new service account the correct permissions when assigned to that account.
+
+After that, you should create a simple YAML file in [the sync-littlepay DAG folder in the data-infra repository](https://github.com/cal-itp/data-infra/tree/8516a1a4ab2ecfe6ef33e3fbc4224bcedbd06e98/airflow/dags/sync_littlepay), structurally similar to the others that are there, utilizing the instance name and access key name for the agency you're setting up. Similarly, you should create a YAML file corresponding to the new agency in [the parse-littlepay DAG folder](https://github.com/cal-itp/data-infra/tree/8516a1a4ab2ecfe6ef33e3fbc4224bcedbd06e98/airflow/dags/parse_littlepay). Then, add a mapping between the agency's instance identifier and its Cal-ITP internal GTFS dataset identifier in [this seed file](https://github.com/cal-itp/data-infra/blob/main/warehouse/seeds/payments_gtfs_datasets.csv).
+
+[This pull request](https://github.com/cal-itp/data-infra/pull/2928/files) demonstrates all four of the code changes involved in this setup process (though the row-level access policy setting is now in a macro [here](https://github.com/cal-itp/data-infra/blob/main/warehouse/macros/create_row_access_policy.sql#L21), rather than in the `fct_payments_rides_v2` model).
+
+After these steps are completed, the new Littlepay account's data should start flowing the next time [the external table Airflow DAG](https://o1d2fa0877cf3fb10p-tp.appspot.com/dags/create_external_tables/grid) runs.
+
+## Rotating Littlepay AWS Keys
+
+Littlepay requests that clients accessing their raw data feeds through S3 rotate the IAM keys every 90 days. They provide general instructions for doing so with the `aws` CLI tool. The following gives additional context on the Cal-ITP setup, and should be used in conjunction with those instructions.
+
+### Create and store new credentials
+
+Secret Manager allows multiple versions of secrets, so create a new version of each secret (the access key ID and the secret key) and paste the new key as the value.
+
+`aws iam create-access-key --user-name <username> --profile <profile>`
+
+You could also use the CLI to create new versions of secrets, but the web UI is very easy to use.
```

Comment:
Can we link to the web UI again? Is there a relevant docs page that can be linked on how to create the new secret version?

Reply:
Resolved in [0049f24](https://github.com/cal-itp/data-infra/pull/3019/commits/0049f24032f29d2a243d2db04fe472dcd4067ae7)